### ISSUE TITLE
There is no way to prepare emscripten HTML output for cgi

### ISSUE THREAD

**Issue by yecril71pl** [emscripten-core/emscripten#14540]
# build
```sh
emcc -ohello.html hello.c
```

## output
```html
<script async type="text/javascript" src="hello.js"></script>
```

# CGI
```sh
transform $resources/hello.html
```

# problem
This CGI script is placed into `cgi-bin` and, accordingly, `hello.js` is not there because it is in *resources*.  It should not be in `cgi-bin` but I cannot find a way to tell `emcc` that it will be placed elsewhere.  I would expect the following result:

## output
```html
<script async type="text/javascript" src="{resources}/hello.js"></script>
```

The same problem applies to `hello.wasm` of course.  Is there a work-around, short of creating an executable script for each resource to stream it?

**Comment by yecril71pl** [comment]
The following workaround does the right thing:
# hello.js
```sh
#!/bin/sh include.sh
```
# include.sh
```sh
case "${1}" in
*.wasm) s=wasm;;
*.js)   s=javascript;;
esac
printf 'Content-Type: application/%s\r\n\r\n' $s
cat "$resources/${1##*/}"
```

**Comment by sbc100** [comment]
Indeed, unless I'm misreading the code, it looks like there is currently a baked in assumption the that JS and HTML files live alongside each other.

My guess as to why other folks might not be running into this is that I think many users switch to writing their own html from scratch once their apps get sufficiently complex/mature and don't use emscripten's html generation at all.  @kripken might have more background on this, and why this feature doesn't already exist.

**Comment by kripken** [comment]
Yes, I think that's it - the default HTML is fairly basic, and good for testing, but in production I believe most people customize it.

**Comment by yecril71pl** [comment]
My (in)ability to customise it is covered in #14541.

**Comment by sbc100** [comment]
So I think this is not a feature we need to add.   Instead folks should look at writing their own html.    I think this can close this can work on #14541 instead?

**Comment by yecril71pl** [comment]
I will take #14541 but I have no assigning powers.  After that one is fixed, this one will lose importance.
